examples/chain-extension/src/chain_extension_ink_side.rs

```diff
@@ -0,0 +1,39 @@
+use ink::env::{chain_extension::FromStatusCode, DefaultEnvironment, Environment};
+
+/// Simple chain extension that provides some staking information.
+#[ink::chain_extension]
+pub trait StakingExtension {
+    type ErrorCode = StakingExtensionErrorCode;
+
+    /// Returns the number of the validators.
+    #[ink(extension = 41)]
+    fn get_num_of_validators() -> u32;
```

Comment:
A fairly minor stuff, but a chain extension should return Result<T, E> by default.
I think with current setup it will counterintuitively still return the  Result<> type.
I believe, to return a u32 you'd need to add #[ink(handle_status = false)] macro on top of the method

https://use.ink/macros-attributes/chain-extension/#details-handle_status

Reply:
yes, it was returning `Result` (see outdated `unwrap` in the contract's message); but I made things explicit in the last commit; thanks for catching this!